File: code/167.两数之和-ii-输入有序数组.cpp
/*
 * @lc app=leetcode.cn id=167 lang=cpp
 *
 * [167] 两数之和 II - 输入有序数组
 */
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int left = 0, right = numbers.size()-1;
        vector<int> res(2);

        while(left < right)
        {
            if(numbers[left] + numbers[right] == target)
            {
                res[0] = left+1;
                res[1] = right+1;
                break;
            }
            else if(numbers[left] + numbers[right] < target)
                ++left;
            else
                --right;
        }
        return res;
    }
};

File: code/167.两数之和-ii-输入有序数组.go
/*
 * @lc app=leetcode.cn id=167 lang=golang
 *
 * [167] 两数之和 II - 输入有序数组
 */

// @lc code=start
func twoSum(numbers []int, target int) []int {
	for i := 0; i < len(numbers); i++ {
		left, right := i+1, len(numbers)-1
		for left <= right {
			mid := left + (right-left) / 2
			if numbers[mid] == target-numbers[i] {
				return []int{i+1, mid+1}
			} else if numbers[mid] < target-numbers[i] {
				left = mid + 1
			} else {
				right = mid - 1
			}
		}
	}
	return []int{-1, -1}
}
// @lc code=end

